protocol/communities/manager.go

```diff
@@ -1562,6 +1590,9 @@ func (m *Manager) HandleCommunityDescriptionMessage(signer *ecdsa.PublicKey, des
 		return nil, err
 	}
 
+	mutex := m.getMutexForCommunity(types.EncodeHex(id))
```

Comment:
I think it's a bit neater if we create a separate object, that keeps it's own state:
```
type CommunityLock struct {
    locks map[string]*sync.Mutex
    mutex sync.Mutex
}
```

and then you can instance it on messenger and provide a couple of methods:

```
  m.communityLocks.LockCommunity(communityID)
  defer m.UnlockCommunity(communityID)
```

In general, you don't want to let consumer access locks directly (albeit this is a bit different, since it's all messenger, but conceptually there's a function gating access to state)